[2023-07-26 05:07:10,744] {taskinstance.py:1179} INFO - Dependencies all met for <TaskInstance: fetch_exchange_rates_dag.fetch_exchange_rates_task scheduled__2023-07-25T22:20:00+00:00 [queued]>
[2023-07-26 05:07:10,761] {taskinstance.py:1179} INFO - Dependencies all met for <TaskInstance: fetch_exchange_rates_dag.fetch_exchange_rates_task scheduled__2023-07-25T22:20:00+00:00 [queued]>
[2023-07-26 05:07:10,763] {taskinstance.py:1376} INFO - 
--------------------------------------------------------------------------------
[2023-07-26 05:07:10,764] {taskinstance.py:1377} INFO - Starting attempt 1 of 1
[2023-07-26 05:07:10,765] {taskinstance.py:1378} INFO - 
--------------------------------------------------------------------------------
[2023-07-26 05:07:10,784] {taskinstance.py:1397} INFO - Executing <Task(PythonOperator): fetch_exchange_rates_task> on 2023-07-25 22:20:00+00:00
[2023-07-26 05:07:10,791] {standard_task_runner.py:52} INFO - Started process 1880 to run task
[2023-07-26 05:07:10,799] {standard_task_runner.py:79} INFO - Running: ['***', 'tasks', 'run', 'fetch_exchange_rates_dag', 'fetch_exchange_rates_task', 'scheduled__2023-07-25T22:20:00+00:00', '--job-id', '491', '--raw', '--subdir', 'DAGS_FOLDER/my_calc_rates.py', '--cfg-path', '/tmp/tmphwx7vq1p', '--error-file', '/tmp/tmp89nxr2fn']
[2023-07-26 05:07:10,802] {standard_task_runner.py:80} INFO - Job 491: Subtask fetch_exchange_rates_task
[2023-07-26 05:07:10,923] {task_command.py:371} INFO - Running <TaskInstance: fetch_exchange_rates_dag.fetch_exchange_rates_task scheduled__2023-07-25T22:20:00+00:00 [running]> on host acb6bca32630
[2023-07-26 05:07:11,011] {taskinstance.py:1591} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=marina_z
AIRFLOW_CTX_DAG_ID=fetch_exchange_rates_dag
AIRFLOW_CTX_TASK_ID=fetch_exchange_rates_task
AIRFLOW_CTX_EXECUTION_DATE=2023-07-25T22:20:00+00:00
AIRFLOW_CTX_TRY_NUMBER=1
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2023-07-25T22:20:00+00:00
[2023-07-26 05:07:11,375] {taskinstance.py:1909} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 171, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/operators/python.py", line 189, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/my_calc_rates.py", line 73, in fetch_exchange_rates
    df = pd.read_csv(f'./csv_files/exchange_latest_{base}.csv', decimal=',', index_col=False)
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/util/_decorators.py", line 311, in wrapper
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/io/parsers/readers.py", line 586, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/io/parsers/readers.py", line 482, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/io/parsers/readers.py", line 811, in __init__
    self._engine = self._make_engine(self.engine)
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/io/parsers/readers.py", line 1040, in _make_engine
    return mapping[engine](self.f, **self.options)  # type: ignore[call-arg]
  File "/home/airflow/.local/lib/python3.7/site-packages/pandas/io/parsers/c_parser_wrapper.py", line 69, in __init__
    self._reader = parsers.TextReader(self.handles.handle, **kwds)
  File "pandas/_libs/parsers.pyx", line 549, in pandas._libs.parsers.TextReader.__cinit__
pandas.errors.EmptyDataError: No columns to parse from file
[2023-07-26 05:07:11,394] {taskinstance.py:1420} INFO - Marking task as FAILED. dag_id=fetch_exchange_rates_dag, task_id=fetch_exchange_rates_task, execution_date=20230725T222000, start_date=20230726T050710, end_date=20230726T050711
[2023-07-26 05:07:11,407] {standard_task_runner.py:97} ERROR - Failed to execute job 491 for task fetch_exchange_rates_task (No columns to parse from file; 1880)
[2023-07-26 05:07:11,451] {local_task_job.py:156} INFO - Task exited with return code 1
[2023-07-26 05:07:11,487] {local_task_job.py:273} INFO - 0 downstream tasks scheduled from follow-on schedule check
